lms/djangoapps/course_home_api/progress/views.py

```diff
@@ -186,8 +187,9 @@ def get(self, request, *args, **kwargs):
         monitoring_utils.set_custom_attribute('user_id', request.user.id)
         monitoring_utils.set_custom_attribute('is_staff', request.user.is_staff)
         is_staff = bool(has_access(request.user, 'staff', course_key))
+        can_masquarade = request.user.has_perm(permissions.CAN_MASQUARADE_LEARNER_PROGRESS, course_key)
```

Comment:
meanwhile @tehreem-sadat , please try to add a test that checks the flow when `permissions.CAN_MASQUARADE_LEARNER_PROGRESS` returns `True` and `False`

Reply:
yess, that can be added, thanks.